use/spatial-stat_SelectionSNP.Rmd

```diff
@@ -0,0 +1,231 @@
+
+qval <- qvalue(x$pvalues)$qvalues 
+alpha <- 0.1
+outliers_pcadapt <- which(qval<alpha)
+print(outliers_pcadapt)
+length(outliers_pcadapt) # 14 outliers
+
+alpha<-0.05 #use of a more stringent threshold to detect outliers
+outliers <- which(qval<alpha)
+print(outliers)
+length(outliers) #14 outliers
+
+```
+
+
+##section 3: outFlank (https://github.com/whitlock/OutFLANK)
+A procedure to find Fst outliers based on an inferred distribution of neutral Fst: it uses likelihood on a trimmed distribution of Fst values to infer the distribution of Fst for neutral marker. This distribution is used to assign q-values to each locus to detect outliers loci potentially due to spatially heterogeneous selection. In practice, first the function MakeDiploidFSTMat allows to  calculate  the appropriate input dataframe among other parameters FST for each loci, and FST without sampling size correction (FSTNoCorr). Then the function outFlank estimate q-values and provides a list of outliers.   
+
+```{r, outFLank}
+
+pop<-data.frame(sel[,1])
+ind<-paste("pop",pop[,1]) #vector with the name of population
+
+locinames <- as.character(seq(1:ncol(genotype))) #vector with the name of loci
+
+FstDataFrame <- MakeDiploidFSTMat(genotype, locinames, ind)
+plot(FstDataFrame$FST, FstDataFrame$FSTNoCorr, xlim=c(-0.01,0.3), ylim=c(-0.01,0.3),pch=20)
+abline(0,1) # Checking the effect of sample size on Fst since FSTCoCorr will be used in the follow
+hist(FstDataFrame$FSTNoCorr) 
+
+OF <- OutFLANK(FstDataFrame, NumberOfSamples=19, qthreshold = 0.05, RightTrimFraction = 0.05)
+
+# Plot the ditribution of Fst with the chi squared distribution
+OutFLANKResultsPlotter(OF, withOutliers = TRUE, NoCorr = TRUE, Hmin = 0.1, binwidth = 0.005, Zoom = FALSE, 
+                       RightZoomFraction = 0.05, titletext = NULL)
+
+outliers_OF <- OF$results$LocusName[OF$results$OutlierFlag==TRUE]
+print(outliers_OF)
+length(outliers_OF)  # 11 outliers
+
+
+```
+
+
+## Section 4: Poisson regression : linking outliers and temperature
+We aim to test if the outliers detected with PCAdapt and OUTFLANCK (663  706  923 1163 1378 1639 1666 1825 2133 2556 2871) are correlated to the temperature. As the raw data are count of alleles, we used a GLM with an error model of poisson. We derive the relation for some of the outliers. You can do the test for all.
+
+
+```{r,Poisson regressions}
+
+# We keep the outliers loci selected by both pcadapt and outflank
+outliers <- outliers_pcadapt[which(outliers_pcadapt %in% outliers_OF)]
+length(outliers) # 11 outliers
+
+loc1<-genotype[,outliers[1]]
+temp<-sel[,2]
+loc1temp<-data.frame(loc1,temp)
+
+m0<-mean(loc1temp[loc1temp$loc1==0,2])
+m1<-mean(loc1temp[loc1temp$loc1==1,2])
+m2<-mean(loc1temp[loc1temp$loc1==2,2])
+boxplot(loc1temp[loc1temp$loc1==0,2],loc1temp[loc1temp$loc1==1,2],loc1temp[loc1temp$loc1==2,2])
+ 
+mod<-glm(loc1~temp,family=poisson)  
+summary(mod)  # This loci is significantly correlated to temperature
+
+ 
+loc2<-genotype[,outliers[2]]

```

Comment:
This section is fairly redundant, it would perhaps be better to show one or two loci and ask the user to do the rest.